contracts/vortex/CarbonVortex.sol

```diff
@@ -0,0 +1,262 @@
+        Token targetToken,
+        uint256 sourceAmount,
+        uint256 minReturnAmount,
+        uint256 deadline,
+        address beneficiary
+    ) external payable returns (uint256);
+}
+
+/**
+ * @dev CarbonVortex contract
+ */
+contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable, Utils {
+    using Address for address payable;
+
+    ICarbonController private immutable _carbonController;
+    IBancorNetwork private immutable _bancorNetwork;
+    Token private immutable _bnt;
+
+    uint256 private _totalBurnt;
+
+    // rewards percentage
+    uint256 private _rewardsPercentagePPM;
+
+    // upgrade forward-compatibility storage gap
+    uint256[MAX_GAP - 2] private __gap;
+
+    /**
+     * @dev a "virtual" constructor that is only used to set immutable state variables
+     */
+    constructor(
+        Token bnt,
+        ICarbonController carbonController,
+        IBancorNetwork bancorNetwork
+    ) validAddress(address(carbonController)) validAddress(Token.unwrap(bnt)) validAddress(address(bancorNetwork)) {
+        _carbonController = carbonController;
+        _bancorNetwork = bancorNetwork;
+        _bnt = bnt;
+    }
+
+    /**
+     * @dev fully initializes the contract and its parents
+     */
+    function initialize() external initializer {
+        __CarbonVortex_init();
+    }
+
+    // solhint-disable func-name-mixedcase
+
+    /**
+     * @dev initializes the contract and its parents
+     */
+    function __CarbonVortex_init() internal onlyInitializing {
+        __Upgradeable_init();
+        __ReentrancyGuard_init();
+
+        __CarbonVortex_init_unchained();
+    }
+
+    /**
+     * @dev performs contract-specific initialization
+     */
+    function __CarbonVortex_init_unchained() internal onlyInitializing {
+        setRewards(100_000);
+    }
+
+    /**
+     * @dev authorize the contract to receive the native token
+     */
+    receive() external payable {}
+
+    /**
+     * @inheritdoc Upgradeable
+     */
+    function version() public pure override(IVersioned, Upgradeable) returns (uint16) {
+        return 2;
+    }
+
+    /**
+     * @inheritdoc ICarbonVortex
+     */
+    function setRewards(uint256 newRewardsPercentagePPM) public onlyAdmin validFee(uint32(newRewardsPercentagePPM)) {
+        uint256 prevRewardsPercentagePPM = _rewardsPercentagePPM;
+
+        // return if the rewards percentage PPM is the same
+        if (prevRewardsPercentagePPM == newRewardsPercentagePPM) {
+            return;
+        }
+
+        _rewardsPercentagePPM = newRewardsPercentagePPM;
+
+        emit RewardsUpdated({ prevRewards: prevRewardsPercentagePPM, newRewards: newRewardsPercentagePPM });
+    }
+
+    /**
+     * @inheritdoc ICarbonVortex
+     */
+    function rewards() external view returns (uint256) {
+        return _rewardsPercentagePPM;
+    }
+
+    /**
+     * @inheritdoc ICarbonVortex
+     */
+    function totalBurnt() external view returns (uint256) {
```

Comment:
Cosmetic: I believe that 'burned' will be consistent with the rest of the code

Reply:
changed

Reply:
There are still a ton of 'burnt' in the code (events, documentation, etc).
Perhaps a find/replace would be better...

Reply:
Changed